### PR TITLE
Prompt Task/Complexity Classifier

### PR DESCRIPTION
**Pull request by sarahyurick**
Awaiting Hugging Face release: https://huggingface.co/nvidia/prompt-task-and-complexity-classifier.

### CODE REVIEW

File: nemo_curator/classifiers/task_complexity.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    This class is optimized for running on multi-node, multi-GPU setups to enable fast and efficient inference on large datasets.
+
+    Attributes:
+        filter_by (list[str], optional): The classes to filter the dataset by.
+                                         If None, all classes will be included. Defaults to None.
+        batch_size (int): The number of samples per batch for inference. Defaults to 256.
+        text_field (str): The field in the dataset that should be classified.
+        # TODO: Clarify output column names
+        pred_column (str): The column name where predictions will be stored. Defaults to "pred".
+        prob_column (str, optional): The column name where prediction probabilities will be stored. Defaults to None.
```
Comment (sarahyurick) [inline nemo_curator/classifiers/task_complexity.py:79]: I am anticipating refactoring the `TaskComplexityClassifier` more, based on our internal model card reference. I currently do not have access to the model itself, so this is mostly placeholder code based off of the `DomainClassifier`.

File: nemo_curator/classifiers/prompt_task_complexity.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+@dataclass
+class PromptTaskComplexityConfig:
+    base_model: str = "microsoft/DeBERTa-v3-base"
+    max_len: int = 512
+    model_output_type: Dict = field(default_factory=dict)
+
+
+class MeanPooling(nn.Module):
+    def __init__(self):
+        super(MeanPooling, self).__init__()
```
Comment (VibhuJawa) [inline nemo_curator/classifiers/prompt_task_complexity.py:45]: In Python>3.0 it is recommended to use super() function without any arguments.
```suggestion
        super().__init__()
```

Detailed explanation around it: 

https://stackoverflow.com/a/27134600

File: nemo_curator/classifiers/prompt_task_complexity.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        sum_mask = input_mask_expanded.sum(1)
+        sum_mask = torch.clamp(sum_mask, min=1e-9)
+
+        mean_embeddings = sum_embeddings / sum_mask
+        return mean_embeddings
+
+
+class MulticlassHead(nn.Module):
+    def __init__(self, input_size, num_classes):
+        super(MulticlassHead, self).__init__()
```
Comment (VibhuJawa) [inline nemo_curator/classifiers/prompt_task_complexity.py:62]:
```suggestion
        super().__init__()
```

File: nemo_curator/classifiers/prompt_task_complexity.py [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+        self.fc = nn.Linear(input_size, num_classes)
+
+    def forward(self, x):
+        x = self.fc(x)
+        return x
+
+
+class CustomHFDeberta(nn.Module, PyTorchModelHubMixin):
+    def __init__(self, config: dataclass):
+        super(CustomHFDeberta, self).__init__()
```
Comment (VibhuJawa) [inline nemo_curator/classifiers/prompt_task_complexity.py:72]:
```suggestion
        super().__init__()
```

File: nemo_curator/classifiers/prompt_task_complexity.py [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+        self,
+        dataset: DocumentDataset,
+    ) -> DocumentDataset:
+        raise NotImplementedError(
+            "filter_by not supported with PromptTaskComplexityClassifier"
+        )
+
+    def get_labels(self):
+        print(
+            "Please see https://huggingface.co/nvidia/prompt-task-and-complexity-classifier for more information about PromptTaskComplexityClassifier outputs."
```
Comment (VibhuJawa) [inline nemo_curator/classifiers/prompt_task_complexity.py:367]: I dont follow this, what are expected labels here ? 

And why are we returning None ?

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Mostly looks good, have requested some minor feedback.

File: nemo_curator/classifiers/prompt_task_complexity.py (hunk 5 above)
Comment (sarahyurick) [reply nemo_curator/classifiers/prompt_task_complexity.py:367]: Since there are 11 output columns I wasn't really sure if we should support this function. Although I guess we could have it return a string or dictionary about the outputs? What do you think?

Comment (VibhuJawa) [reply nemo_curator/classifiers/prompt_task_complexity.py:367]: Dont think we should support it. You should raise a NotImplementedError here with the error message.

**Review by VibhuJawa — APPROVED** [review]
LGTM
